capstone-frontend/src/pages/BorderPage.jsx

```diff
@@ -44,62 +43,41 @@ function BorderPage() {
       .catch((error) => console.error(error));
   }
 
-  function createCommentDisplays(specificBorderPosts) {
-    allRelevantComments = [];
-    specificBorderPosts.forEach((element) => {
-      console.log(element);
-      // if (parseInt(element.borderNum) == borderIndex) {
-      //   allRelevantComments.push(element.userInput);
-      // }
-    });
-    return specificBorderPosts;
-  }
   const { data, error, isLoading } = useSWR(
     "http://localhost:5000/borderdata",
     fetcher
   );
   const crossingData = data;
 
-  useEffect(() => {}, [userComments]);
-
-  let position;
-  // let latitude = position.lat;
-  // let longitude = position.lng;
-  // position = { lat: latitude, lng: longitude };
   const fetchCoordinates = async (address) => {
     const apiKey = "AIzaSyDnk1NQgt08aY9-4tS0ZcG9WvzJc7hsuWE";
-    const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${encodeURIComponent(address)}&key=${apiKey}`;
-    const response = await fetch(url)
+    const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${encodeURIComponent(
+      address
+    )}&key=${apiKey}`;
+    const response = await fetch(url);
     const coordinateData = await response.json();
     const newCoords = await coordinateData.results[0].geometry.location;
     setCoordinates(newCoords);
     return newCoords;
   };
-  const {
-    data: data2,
-    error: error2,
-    isLoading: isLoading2,
-  } = useSWR("http://localhost:5000/usersposts", fetcher);
-  let allComments = data2;
 
-  // createCommentDisplays(allComments)
-  // createCommentDisplays(allComments)
-  // console.log(createCommentDisplays(allComments))
-
-  const fetchIndividualImage = async (location) => {
-    const response = await fetch(
-      `https://images-api.nasa.gov/search?q=${location}&media_type=image`
-    );
-    const data = await response.json();
-    const newImage = await data.collection.items[0].links[0].href;
-    setImageUrl(newImage);
-  };
+  useEffect(() => {
+    fetch("http://localhost:5000/usersposts")
+      .then((response) => response.json())
+      .then((data) => {
+        let relevantComments = [];
+        for (const item in data) {
+          if (data[item].borderNum == borderIndex) {
+            relevantComments.push(data[item]);
+          }
```

Comment:
Filter on the server if you can

Reply:
Made this a separate component for readability. I did this instead of putting it on the server as I need to sort my comments by borderIndex and by userId. I supposed I could have a server call for each as well.